### DISCUSSION TITLE
CAIP codes decode

### DISCUSSION THREAD

**Discussion by 2jumper3** [WalletConnect/walletconnect-monorepo#3873] [Developer Support]
Hi all!

I'm trying to understand how to get the correct address to provide it in the `Account` struct when you only have the CAIP code.

I think I might be on the wrong path because, in my opinion, it currently looks like:

```swift
if chain.absoluteString == "eip155:1" ||
    chain.absoluteString == "eip155:5" ||
    chain.absoluteString == "eip155:8453" {
    let address = coins.first { coin in
        coin.network.id == "ETH"
    }!
    let coinAddress = WalletService.shared.wallet.getAddressForCoin(coin: address.network.addressType)
    return WalletConnectSign.Account(blockchain: chain, address: coinAddress)!
} else if chain.absoluteString.contains("tron") {
    let address = coins.first { coin in
        coin.network.id == "TRX"
    }!
    let coinAddress = WalletService.shared.wallet.getAddressForCoin(coin: address.network.addressType)
    return WalletConnectSign.Account(blockchain: chain, address: coinAddress)!
}
```

I've hardcoded these CAIP codes and provided the chain code for each one. But if there are 100 codes, it will be a very huge `if else` method :-)

How do I compare the CAIP code with the chain to understand which address I should send in the `Account` struct, or how do I do it correctly?

**Comment by glitch-txs** [comment]
hi, which library are you using? could you share a full example of the issue/question? public repository or a codesandbox like example

**Reply by 2jumper3** [reply]
Hi @glitch-txs ! 
Web3Wallet. 
I use this manual: https://docs.walletconnect.com/web3wallet/wallet-usage#setup-auth-requests-handling 
<img width="971" alt="Screenshot 2023-11-08 at 16 35 30" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/45697468/0262f0d2-5209-4cd0-84b2-c11d42df0c5e">

**Reply by glitch-txs** [reply]
we'll make braking changes to Auth so it's not recommended to be integrated rn

**Reply by 2jumper3** [reply]
Hi @glitch-txs, and many thanks for the fast reply! Does this mean that we should stop developing Web3Wallet and wait for some updates? Could you please provide an estimated time for the new features?

**Reply by glitch-txs** [reply]
You can integrate Web3Wallet for connecting, signing etc. We'll change the Auth part

**Reply by 2jumper3** [reply]
But this is already Web3Wallet. I've read the documentation for Web3Wallet in your guidelines at https://docs.walletconnect.com/web3wallet/wallet-usage#setup-auth-requests-handling.

Where can I find the new documentation? Because I'm following these docs step by step, and I'm a little bit confused. 😅

**Reply by glitch-txs** [reply]
We removed auth from the docs
